### ISSUE TITLE
Adding a UIActivityIndicatorView?

### ISSUE THREAD

**Issue by schmorgledorf** [youtube/youtube-ios-player-helper#48]
Obviously, the video doesn't snap in right off the bat. Should there be a UIActivityIndicator that takes the blank space until the webView pops in? Where would that go?

**Comment by JALsnipe** [comment]
You have an activity indicator that starts when you initialize the player and end when the player transitions to the playing state using the delegate.

Eg:
```
- (void)viewDidLoad {
    if([self loadWithVideoId:[VIDEO_ID]) {
            self.castView.delegate = self;
            // start your UIActivityIndicator here
        } else { // failure
            NSLog(@"failed loading player");
        }
}
...
- (void)playerView:(YTPlayerView *)playerView didChangeToState:(YTPlayerState)state
{
    NSLog(@"YTPlayerView : didChangeToState");

    switch (state) {
        case kYTPlayerStatePlaying:
            NSLog(@"Started playback");
            // stop UIActivityIndicator here
            break;
        case kYTPlayerStatePaused:
            NSLog(@"Paused playback");
            break;
        case kYTPlayerStateEnded:
                NSLog(@"Video ended");
                break;
        case kYTPlayerStateUnstarted:
            NSLog(@"Player in Unstarted State");
        default:
            break;
    }
}
```

**Comment by nikkumang** [comment]
@JALsnipe Good point. I used playerViewDidBecomeReady:(YTPlayerView *)playerView to stop the activity indicator.

**Comment by Gloix** [comment]
I'm trying to do this but it doesn't seem to work well. If I try it with a good internet connection the indicator shows up for a brief time before the video loads. 

On the other hand, if I try with wifi disconnected, the indicator also shows up for a brief time, but then I'm left with a black rectangle (the player) and I receive no error in `playerView(_ playerView: YTPlayerView, receivedError error: YTPlayerError)`, nor is `playerViewDidBecomeReady(_ playerView: YTPlayerView)` called, which is where the activity indicator is stopped.

Also, the player is left in the `unknown` state, which does not give me much information, since the unknown state is actually part of the normal initialization process of the player. Indeed, I receive two or three `unknown` state changes before the player becomes ready when using a stable internet connection.


The weird thing about the activity indicator that disappears after a brief time when offline is that if I debug the view hierarchy, I can see the activity indicator, but it doesn't show up on screen.
